math/pronicnumber.go

```diff
@@ -0,0 +1,21 @@
+// pronicnumber.go
+// description: Returns true if the number is pronic and false otherwise
+// details:
+//  Pronic number: For any integer n, if there exists integer m
+//  such that n = m * (m + 1) then n is called a pronic number.
```

Comment:
```suggestion
//  Pronic number: A number n is pronic, if there
//  exists an integer m such that n = m * (m + 1)
```
I am not sure about the proper indentation here - @tjgurwara99  knows for sure how it should be done exactly.
What do you think about being consistent and using the same symbol for the _number that exists_? Now in the definition `m` is used and in the implementation there is `x`.

Reply:
The way described above was done based on my suggestion earlier - which is the proper way to define mathematical definitions in proper logical order. Though arguably English is ambiguous and that order also makes sense.

Reply:
There is a subtle difference between the proposed definitions. If you would have an exercise about proving that a given number is pronic and the "current definition" is used, then you have to:
- show that your number is an integer,
- show that it is a product of two consecutive integers.

_Obviously_ the first statement follows from the second one. But, in order to be fully correct, you still have to do it. Therefore I suggest to omit the _requirement_ that a pronic number is an integer.

Moreover using the quantifier "For any n..." makes the thing much more complicated.

Reply:
I don't know what you mean by "current definition" which one are you referring to? I think you're referring to your definition to prove your point but lets see.

The exercise "if you would have an exercise about proving that a given number is pronic" could just as easily be changed to "If you would have an exercise about proving that a given integer is pronic". Now you don't need to prove it and if you were to look closely at the function definition, the priori of the number being an integer is already there, that is, it only accepts integers.

> Obviously the first statement follows from the second one. 

Yes, and it is trivially so... unless you're proving the integer's are closed under multiplication

> Moreover using the quantifier "For any n..." makes the thing much more complicated.

Depends, but lets not go into that in this discussion

Reply:
> I don't know what you mean by "current definition" which one are you referring to? I think you're referring to your definition to prove your point but lets see.

I was referring to:
```
//  Pronic number: For any integer n, if there exists integer m
//  such that n = m * (m + 1) then n is called a pronic number.
```
Going in this direction one could also write thing like:
```
//  Pronic number: For any positive and even integer n greater or equal 2, if there exists integer m
//  such that n = m * (m + 1) then n is called a pronic number.
```
Therefore I suggest [this](https://github.com/TheAlgorithms/Go/pull/595#discussion_r1017509813).

Reply:
Hahah, please read the part about function definition 😄 it isn't `func PronicNumber(n any)` it is `func PronicNumber(n int)`

Reply:
> Hahah, please read the part about function definition 😄 it isn't `func PronicNumber(n any)` it is `func PronicNumber(n int)`

[We already discussed that](https://github.com/TheAlgorithms/Go/pull/595#discussion_r1006597506). It should be `func PronicNumber(n uint)`. Jokes aside: you are mixing two things here: the documentation is _for the human_ and the _function signature_ is for _for the computer_ (and for the human). Very roughly speaking any definition is easier to understand/work with if there are fewer requirements.



> If you want to be combative about things and your suggestions, I would suggest you don't do that here 🙏🏼

I am faraway of doing that. Suggestion is just a suggestion. If it will not be accepted, well I do not care. I will rethink it, think about the reasons why it was not accepted and next time I will make a better one.

Reply:
I like this conversation, let's move this to a [discussion](https://github.com/TheAlgorithms/Go/discussions/605). If needed we can make changes to the docs in a later PR - let's not block it for now. 😄